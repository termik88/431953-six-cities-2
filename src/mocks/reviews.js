const reviews = [
  {
    id: 1,
    user: {
      id: 15,
      name: `Kendall`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/6.jpg`,
      isPro: false
    },
    rating: 80,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`
  },
  {
    id: 2,
    user: {
      id: 14,
      name: `Corey`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/5.jpg`,
      isPro: true
    },
    rating: 60,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`
  },
  {
    id: 3,
    user: {
      id: 19,
      name: `Christina`,
      avatarUrl: `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/10.jpg`,
      isPro: false
    },
    rating: 60,
    date: `2019-12-04T09:09:49.938Z`,
    commentText: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`
  }
];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export default reviews;
