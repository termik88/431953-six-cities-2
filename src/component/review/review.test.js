import React from 'react';
import renderer from 'react-test-renderer';

import Review from "./review.jsx";
import reviews from "../../mocks/reviews.js";

describe(`Review component test.`, () => {
  it(`Correctly renders after relaunch`, () => {
    const mockDate = {
      placeId: 1,
      comments: reviews,
      isAuthorizationRequired: false,
      isLoading: false,
      errorInfo: ``,
      onSendComment: jest.fn(),
    };

    const test = renderer
      .create(
          <Review
            {...mockDate}
          />)
      .toJSON();

    expect(test).toMatchSnapshot();
  });
});
