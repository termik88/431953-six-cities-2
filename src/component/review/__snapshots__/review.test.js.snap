// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review component test. Correctly renders after relaunch 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â·
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Review avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/6.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Kendall
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          December 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Review avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/5.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Corey
        </span>
        <span
          className="reviews__user-status"
        >
          Pro
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          December 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Review avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/10.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Christina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          December 2019
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-stars"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value=""
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Submit
      </button>
    </div>
    
  </form>
</section>
`;
